apps/user/src/main/java/com/ecommerce/user/controller/UserController.java

```diff
@@ -0,0 +1,30 @@
+package com.ecommerce.user.controller;
+
+import com.ecommerce.user.application.request.SignUpRequest;
+import com.ecommerce.user.application.response.GenericResponse;
+import com.ecommerce.user.service.UserAccountService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@Slf4j
+@Validated
+@RequiredArgsConstructor
+@RestController
+public class UserController {
+    private final UserAccountService userAccountService;
+
+    @PostMapping("/signup")
+    public ResponseEntity<GenericResponse> signUpNewUser(
+            @Valid @RequestBody final SignUpRequest signUpRequest
```

Comment:
(Optional) Date of birth regex validation rule
![image](https://github.com/user-attachments/assets/d2d0a89c-4480-434e-ba0e-7d81add5a924)